MapboxDirections/OfflineDirections.swift

```diff
@@ -0,0 +1,139 @@
+import Foundation
+
+
+public typealias OfflineDownloaderCompletionHandler = (_ location: URL?,_ response: URLResponse?, _ error: Error?) -> Void
+public typealias OfflineDownloaderProgressHandler = (_ bytesWritten: Int64, _ totalBytesWritten: Int64, _ totalBytesExpectedToWrite: Int64) -> Void
+public typealias OfflineVersionsHandler = (_ version: [Version]?, _ error: Error?) -> Void
+
+struct AvailableVersionsResponse: Codable {
+    let availableVersions: [Version]
+}
+
+@objc(MBVersion)
+public class Version: NSObject, Codable {
+    let versionString: String
+    let versionDate: Date
+    
+    static var dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd"
```

Comment:
On the off-chance that there needs to be two versions on the same day, would the API ever assign a more precise timestamp?